### ISSUE TITLE
warning from swift demo 

### ISSUE THREAD

**Issue by dakeshi** [firebase/FirebaseUI-iOS#166]
I download and pod install for swift demo and got the following warning in multiple files:
Conflicting nullability specifier on return types, 'nullable' conflicts with existing specifier 'nonnull'

FIRApp.h:71:1: 
FIREmailPasswordAuthProvider.h:35:1: 
FIRFacebookAuthProvider.h:34:1: 
FIRGoogleAuthProvider.h:36:1: 
FIRAuth.h:97:1: 
FIRAuthCredential.h:27:1: 
FIRUser.h:72:1:
FIRUser.h:263:1: 
FIRAuthUIBaseViewController.h:48:1: 
FIRAuthUI.h:117:1: 

It seems that NS_ASSUME_NONNULL_BEGIN and nullable make a trouble in the header file.

### Describe your environment

  * Swift: 3.0
  * iOS version: ios 10.0.2
  * FirebaseUI version: 0.6.1
  * CocoaPods Version: 1.0.1

**Comment by morganchen12** [comment]
Many of these are in upstream Firebase (FIRAuth.h, FIRAuthCredential.h, FIRUser.h, FIRApp.h). Fixing these is as easy as deleting the `nullable` annotations on the offending methods.

**Comment by dakeshi** [comment]
In the case of FIRAuth, the following method makes a warning:

```objc
- (nullable instancetype)init NS_UNAVAILABLE;
```

I agree that it can be solved with deleting `nullable` annotations from `init` method.

**Comment by morganchen12** [comment]
Fixed in 633cebcbeae5fb15edef13f4c6381d6554f0a563. There's a Google-internal change to fix the nullability specifiers in upstream Firebase as well.

I'm going to close this issue now since there's no more actions to be taken, but keep in mind you won't see the fix until both Firebase and FirebaseUI make new releases.

**Comment by warby613** [comment]
Rather than deleting the `nullable` annotation, a more passive workaround is to simply suppress the nullability warning for the offending line using `#pragma` directives:

```objective-c
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wnullability"
- (nullable instancetype)init NS_UNAVAILABLE;
#pragma clang diagnostic pop

**Comment by 6ewis** [comment]
@morganchen12 Instead of waiting for the new releases - is that appropriate to change those lines to "(instancetype)init NS_UNAVAILABLE;" as per your fix `633cebc`

**Comment by morganchen12** [comment]
@6ewis the new releases have been out for some time now; if you're still seeing these warnings try running `pod update` and clearing out DerivedData/CocoaPods cache.
